### DISCUSSION TITLE
Bound mutate doesn't trigger SWRConfig onError

### DISCUSSION THREAD

**Discussion by andersnylund** [vercel/swr#1575] [Q&A]
I am trying to create a "global" error handling so that if either an useSWR or an mutate fails a notification will be shown. However, I am not able to get the bound mutate to cause an error that would be caught be the onError-handler in the SWRConfig that wraps the application. 

I have a reproduction here: https://codesandbox.io/s/elegant-herschel-n59qk  

As you can see the error thrown inside the fetcher in useSWR will cause the onError handler to log the error in the console. However the error thrown inside the mutate is not showing up in onError. However the error-object returned from useSWR call is returning the error that was caused by the mutate.

Is it possible to get the errors from bounded mutations to show up in onError-handler?

**Comment by agatsumayuki** [comment]
I have the same problem here. Is there any solutions to this problem?

**Comment by cellulosa** [comment]
any hint?

**Comment by piassi** [comment]
+1 also having this issue at the moment.

**Comment by daveycodez** [comment]
Sheittttt guess I gotta make a custom SWR Error Context implementation ;/

EDIT :

My solution is just to add

```
const { onError } = useSWRConfig()
```

Then

```
try { mutate() } catch (error) { onError(error) }
```

Kinda janky and not ideal D: maybe put it directly into your fetcher for mutates and catch it there.
